src/tensorlake/function_executor/service.py

```diff
@@ -256,3 +364,172 @@ def get_info(
+                "task execution failed in background thread",
+                task_id=task_execution.task.task_id,
+                exc_info=e,
+            )
+
+        finally:
+            # Store TaskResult directly in task
+            with task_execution.updated:
+                task_execution.task.result.CopyFrom(result)
+                task_execution.updated.notify_all()
+
+    def list_tasks(
+        self, request: ListTasksRequest, context: grpc.ServicerContext
+    ) -> ListTasksResponse:
+        with self._tasks_lock:
+            tasks = []
+            for te in self._tasks.values():
+                with te.updated:
+                    # Create a copy of the task without the request field for listing
+                    task_copy = Task()
+                    task_copy.CopyFrom(te.task)
+                    task_copy.ClearField(
+                        "request"
+                    )  # Don't return input data when listing
+                    tasks.append(task_copy)
+
+            return ListTasksResponse(tasks=tasks)
+
+    def create_task(
+        self, request: CreateTaskRequest, context: grpc.ServicerContext
+    ) -> Task:
+        task = request.task
+
+        self._validate_task(task)
+
+        with self._tasks_lock:
+            if task.task_id in self._tasks:
+                context.abort(
+                    grpc.StatusCode.ALREADY_EXISTS,
+                    f"Task {task.task_id} already exists",
+                )
+
+            task_execution = _TaskExecution(task)
+            self._tasks[task.task_id] = task_execution
+
+            task_execution.thread = threading.Thread(
+                target=self._execute_task_in_thread,
+                args=(task_execution,),
+                daemon=True,
+            )
+            task_execution.thread.start()
+
+        # Return a minimal task with no optional fields set; we'll
+        # extend this iff the server provides new info.
+        return Task()
+
+    def await_task(
+        self, request: AwaitTaskRequest, context: grpc.ServicerContext
+    ) -> Generator[AwaitTaskProgress, None, None]:
+        """Wait for task completion and stream progress updates."""
+        task_id = request.task_id
+
+        with self._tasks_lock:
+            task_execution = self._tasks.get(task_id)
+
+        if task_execution is None:
+            context.abort(
+                grpc.StatusCode.NOT_FOUND,
+                f"Task {task_id} not found",
+            )
+
+        # Stream progress updates until task completes
+        final_result = None
+        while True:
+            with task_execution.updated:
+                if (
+                    task_execution.task.result.outcome_code
+                    != TaskOutcomeCode.TASK_OUTCOME_CODE_UNKNOWN
+                ):
+                    # Use TaskResult directly
+                    final_result = task_execution.task.result
+                    break
+
+                task_execution.updated.wait()
```

Comment:
Would this result in Executor not getting a progress update if it was added while Executor was reconnecting to await_task?

Reply:
Done - we'll now always immediately report _something_.